common/auto-updater/http-downloader.h

```diff
@@ -0,0 +1,35 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2020 Intel Corporation. All Rights Reserved.
+
+#pragma once
+
+#include <string>
+#include <functional>
+#include <vector>
+
+namespace rs2
+{
+    enum class callback_result {CONTINUE_DOWNLOAD, STOP_DOWNLOAD};
+    typedef std::function<callback_result(uint64_t dl_current_bytes, uint64_t dl_total_bytes)> user_callback_func_type;
```

Comment:
I don't like this in the general rs2 namespace... can we place everything auto-update-related in a common namespace, maybe, like rs2::auto_updater?

Reply:
1. You are right , I also prefer to give it a unique namespace.
2. Same as previous comment, maybe http-downloader and the auto-update class should be in a separate namespace?

What do you think?